modules/account-lib/src/coin/cspr/transactionBuilder.ts

```diff
@@ -120,7 +212,25 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
   }
 
   validateMandatoryFields(): void {
-    throw new NotImplementedError('validateMandatoryFields not implemented');
+    if (this._fee === undefined) {
```

Comment:
Split this in two little functions, validate fee and validate source and then call it into validate mandatory fields.